src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -0,0 +1,294 @@
+        """
+        The content of a descriptor transformed into a format ready to create a
+        ManifestEntry object.
+        """
+        return {
+            'content-type': self.content['content_type'],
+            'crc32c': self.content['crc32c'],
+            'indexed': True,
+            'name': self.content['file_name'],
+            's3_etag': self.content['s3_etag'],
+            'sha1': self.content['sha1'],
+            'sha256': self.content['sha256'],
+            'size': self.content['size'],
+            'uuid': self.content['file_id'],
+            'version': self.content['file_version']
+        }
+
+
+@dataclass
+class StagingArea:
+    links: Mapping[str, LinksFile]
+    metadata: Mapping[str, MetadataFile]
+    descriptors: Mapping[str, DescriptorFile]
+
+    def file_ids_by_entity_type(self,
+                                subgraph_id: str
+                                ) -> Mapping[str, Set[str]]:
+        """
+        Parse the links in a subgraph and return a mapping of the metadata file
+        types (e.g. 'analysis_file', 'cell_suspension') to a set of file IDs
+        """
+        links_file: LinksFile = self.links[subgraph_id]
+        links_json = links_file.content
+        file_ids = defaultdict(set)
+        # Project ID is only mentioned in the links JSON if there is a
+        # supplementary_file_link so add it in here to make sure it is included.
+        file_ids['project'].add(links_file.project_id)
+        link: JSON
+        for link in links_json['links']:
+            link_type = link['link_type']
+            if link_type == 'process_link':
+                file_type = link['process_type']
+                file_id = link['process_id']
+                file_ids[file_type].add(file_id)
+                for process_link_type in ('input', 'output', 'protocol'):
+                    for file in link[f'{process_link_type}s']:
+                        file_type = file[f'{process_link_type}_type']
+                        file_id = file[f'{process_link_type}_id']
+                        file_ids[file_type].add(file_id)
+            elif link_type == 'supplementary_file_link':
+                for file in link['files']:
+                    file_type = file['file_type']
+                    file_id = file['file_id']
+                    file_ids[file_type].add(file_id)
+            else:
+                raise ValueError(f'Unknown link type: {link_type}')
+        return dict(file_ids)
+
+    def get_bundle_metadata(self, subgraph_id: str) -> Tuple[str, List[JSON], JSON]:
+        """
+        Return a :py:class:`humancellatlas.data.metadata.api.Bundle` object for
+        the given subgraph ID.
+        """
+        logger.debug('Composing bundle %s', subgraph_id)
+
+        links_file = self.links[subgraph_id]
+        manifest = []
+        metadata = {
+            'links.json': links_file.content
+        }
+        file_type_ids = self.file_ids_by_entity_type(subgraph_id)
```

Comment:
When I request a rename, I would appreciate it if you embraced that request and apply it not just literally (and minimally) but also in spirit. What you call `file_type` is what we usually call `entity_type`. So either `entity_type` or `file_entity_type`. And what you call `file_id` is not the UUID of a data file (which is what we usually refer to as `file_id`) but an `entity_id`. So it should be called either that or `file_entity_id`.